compact.go

```diff
@@ -186,19 +191,18 @@ func (c *LeveledCompactor) plan(dms []dirMeta) ([]string, error) {
 		return res, nil
 	}
 
-	// Compact any blocks that have >5% tombstones.
+	// Compact any blocks with big enough time range that have >5% tombstones.
 	for i := len(dms) - 1; i >= 0; i-- {
 		meta := dms[i].meta
 		if meta.MaxTime-meta.MinTime < c.ranges[len(c.ranges)/2] {
-			break
+			continue
```

Comment:
https://github.com/prometheus/tsdb/pull/374/files#diff-43297cfd6441b52c54c50d646a564339R178

The above ensures that we can break here. Basically we're going from big blocks to small blocks and if we find a small block, we can be assured that further blocks will be the same size if not smaller.